### ISSUE TITLE
Installation error: undefined local variable or method `blogs' in User model

### ISSUE THREAD

**Issue by jfdimark** [Bodacious/blogit#15]
Hi - I've followed the installation process and it all seemed really simple, but when I try to start the server it simply won't accept 'blogs' as a method it recognises in the User model.

The full error is pasted below - anyone else had this problem?  Any idea what could be causing it?

I have an STI model setup with 2 child's of User, but its not caused me any other issues in months of development; and I already has acts_as_taggable installed, so it skipped one of the migrations because it was called the same thing (Migration 20110814103306_acts_as_taggable_on_migration.rb from blogit has been skipped. Migration with the same name already exists).

They are the only non-standard things I can think that would be causing an issue.

Thanks for any help...
Mark

```
/Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activerecord-3.2.11/lib/active_record/dynamic_matchers.rb:55:in `method_missing': undefined local variable or method `blogs' for #<Class:0x007f9cdc89a460> (NameError)
    from app/models/user.rb:52:in `<class:User>'
    from app/models/user.rb:49:in `<top (required)>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:469:in `load'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:469:in `block in load_file'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:639:in `new_constants_in'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:468:in `load_file'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:353:in `require_or_load'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:313:in `depend_on'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:225:in `require_dependency'
    from /Users/markwalker/Xavy2/config/initializers/preload_sti_models.rb:3:in `block in <top (required)>'
    from /Users/markwalker/Xavy2/config/initializers/preload_sti_models.rb:2:in `each'
    from /Users/markwalker/Xavy2/config/initializers/preload_sti_models.rb:2:in `<top (required)>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:245:in `load'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:245:in `block in load'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:236:in `load_dependency'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:245:in `load'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/engine.rb:588:in `block (2 levels) in <class:Engine>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/engine.rb:587:in `each'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/engine.rb:587:in `block in <class:Engine>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/initializable.rb:30:in `instance_exec'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/initializable.rb:30:in `run'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/initializable.rb:55:in `block in run_initializers'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/initializable.rb:54:in `each'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/initializable.rb:54:in `run_initializers'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/application.rb:136:in `initialize!'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/railtie/configurable.rb:30:in `method_missing'
    from /Users/markwalker/Xavy2/config/environment.rb:5:in `<top (required)>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:251:in `require'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:251:in `block in require'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:236:in `load_dependency'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/activesupport-3.2.11/lib/active_support/dependencies.rb:251:in `require'
    from /Users/markwalker/Xavy2/config.ru:3:in `block in <main>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/builder.rb:51:in `instance_eval'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/builder.rb:51:in `initialize'
    from /Users/markwalker/Xavy2/config.ru:in `new'
    from /Users/markwalker/Xavy2/config.ru:in `<main>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/builder.rb:40:in `eval'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/builder.rb:40:in `parse_file'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/server.rb:200:in `app'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/commands/server.rb:46:in `app'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/server.rb:304:in `wrapped_app'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/rack-1.4.3/lib/rack/server.rb:254:in `start'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/commands/server.rb:70:in `start'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/commands.rb:55:in `block in <top (required)>'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/commands.rb:50:in `tap'
    from /Users/markwalker/.rvm/gems/ruby-1.9.3-p327/gems/railties-3.2.11/lib/rails/commands.rb:50:in `<top (required)>'
    from script/rails:6:in `require'
    from script/rails:6:in `<main>'
```

**Comment by jfdimark** [comment]
ps.  I've now just added the line   has_many :blog_posts, :as => "blogger", :class_name => "Blogit::Post" and everything seems to work fine now, although 'blogs' doesn't work.

**Comment by Bodacious** [comment]
@jfdimark Thanks - I'll reopen this and have a look at the issue this evening

**Comment by Bodacious** [comment]
@jfdimark - OK, obvious things first:
- User inherits from ActiveRecord::Base
- You've restarted the server after installing blogit

Other than that, I can't think of any reason why the method wouldn't work, it's basically just calling the same code you've pasted above (`has_many :blog_posts, :as => "blogger", :class_name => "Blogit::Post"`)

Can you start up a console session and tell me what the result of this is:

``` ruby
User.included_modules.include?(Blogit::Blogs)
```

... it _should_ return true.

**Comment by jfdimark** [comment]
Hi 
Yep - User inherits from ActiveRecord::Base and I restarted the server (several times) AND the console returns true.
Strange - I seem to be a magnet for odd rails behaviours :-)
Cheers for looking into it.
Mark

Date: Tue, 29 Jan 2013 10:34:07 -0800

**Comment by Bodacious** [comment]
@jfdimark  - could you please try installing this app and let me know if that works OK?

https://github.com/Bodacious/BlogitTestApplication

Some other things to check:

``` ruby
User.respond_to?(:blogs) # returns true?
```

How about if you `include Blogit::Blogs` directly within your User model - does the `blogs` method work after that?

If the TestApplication works in your environment, but the above doesn't, then I guess it could be an issue with another Gem? In that case - could you paste your Gemfile?

Thanks

**Comment by jfdimark** [comment]
Hi Gavin,
I tried everything below, and then as normal, and today it works.
I'm not 100% what has changed, but I was also having issues with active_admin yday due to conflict with redcloth, which seems to be a gem dependency of blogit too.  I resolved that, and I think it must have helped resolved the blogit issue too.
Anyway, thanks for your support and an awesome gem, it's all running perfectly and looks great!
Cheers,Mark

Date: Wed, 30 Jan 2013 03:49:38 -0800
